code/datums/components/fullauto.dm

```diff
@@ -230,7 +230,8 @@
 		stop_autofiring() //Elvis has left the building.
 		return FALSE
 	shooter.face_atom(target)
-	COOLDOWN_START(src, next_shot_cd, autofire_shot_delay)
+	var/next_delay = autofire_shot_delay * (HAS_TRAIT(shooter, TRAIT_DOUBLE_TAP) ? 0.5 : 1)
```

Comment:
you sure about this
I also wish there was a cleaner way to do this

Reply:
I'm sure it'll be fiiine balance wise
also I killed the tertiary check since yeah it was kinda ugly